app/code/Magento/CatalogInventory/Model/Product/QuantityValidator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright 2024 Adobe
+ * All Rights Reserved.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogInventory\Model\Product;
+
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+
+class QuantityValidator
+{
+    /**
+     * @var StockRegistryInterface
+     */
+    private $stockRegistry;
+
+    /**
+     * @param StockRegistryInterface $stockRegistry
+     */
+    public function __construct(
+        StockRegistryInterface $stockRegistry
+    ) {
+        $this->stockRegistry = $stockRegistry;
+    }
+
+    /**
+     * To get quantity validators
+     *
+     * @param int $productId
+     * @param int $websiteId
+     *
+     * @return array
+     */
+    public function getData($productId, $websiteId): array
+    {
+        $stockItem = $this->stockRegistry->getStockItem($productId, $websiteId);
+
+        $params = [];
+        $validators = [];
+        $params['minAllowed'] = (float)$stockItem->getMinSaleQty();
```

Comment:
`getMinSaleQty` is returning the `float` value. No need to typecast here.

Reply:
removed the typecast